relentless/simulate/lammps.py

```diff
@@ -0,0 +1,361 @@
+import abc
+import os
+
+import numpy as np
+
+from relentless.core.volume import TriclinicBox
+from . import simulate
+
+try:
+    import lammps
+    _lammps_found = True
+except ImportError:
+    _lammps_found = False
+
+class LAMMPS(simulate.Simulation):
+    def __init__(self, operations, quiet=True, **options):
+        if not _lammps_found:
+            raise ImportError('LAMMPS not found.')
+
+        super().__init__(operations,**options)
+        self.quiet = quiet
+
+    def _new_instance(self, ensemble, potentials, directory):
+        sim = super()._new_instance(ensemble,potentials,directory)
+
+        if self.quiet:
+            # create lammps instance with all output disabled
+            launch_args = ['-echo','none',
+                           '-log','none',
+                           '-screen','none',
+                           '-nocite']
+        else:
+            launch_args = ['-echo','screen',
+                           '-log', sim.directory.file('log.lammps'),
+                           '-nocite']
+        sim.lammps = lammps.lammps(cmdargs=launch_args)
+
+        # lammps uses 1-indexed ints for types, so build mapping in both direction
+        sim.type_map = {}
+        sim.typeid_map = {}
+        for i,t in enumerate(sim.ensemble.types):
+            sim.type_map[t] = i+1
+            sim.typeid_map[sim.type_map[t]] = t
+
+        return sim
+
+class LAMMPSOperation(simulate.SimulationOperation):
+    fix_id = 1
+
+    def __call__(self, sim):
+        cmds = self.to_commands(sim)
+        sim.lammps.commands_list(cmds)
+
+    @classmethod
+    def add_fix(cls):
```

Comment:
Let's rename this to make it clear what the method does:
```suggestion
    def new_fix_id(cls):
```